java/vector/src/test/java/org/apache/arrow/vector/compare/TestRangeEqualsVisitor.java

```diff
@@ -282,6 +282,48 @@ public void testUnionVectorRangeEquals() {
     }
   }
 
+  /**
+   * Test comparing two union vectors.
+   * The two vectors are different in total, but have a range with equal values.
+   */
+  @Test
+  public void testUnionVectorSubRangeEquals() {
+    try (final UnionVector vector1 = new UnionVector("union", allocator, null, null);
+         final UnionVector vector2 = new UnionVector("union", allocator, null, null);) {
+
+      final NullableUInt4Holder uInt4Holder = new NullableUInt4Holder();
+      uInt4Holder.value = 10;
+      uInt4Holder.isSet = 1;
+
+      final NullableIntHolder intHolder = new NullableIntHolder();
+      uInt4Holder.value = 20;
```

Comment:
was this intended to set intHolder?

Reply:
You are right. Sorry about my mistake. Resolved now.